skglm/solvers/prox_newton_solver.py

```diff
@@ -0,0 +1,225 @@
+import numpy as np
+from scipy import sparse
+from numba import njit
+from skglm.datafits import Logistic, Logistic_32
+from skglm.solvers.common import construct_grad
+from skglm.utils import AndersonAcceleration
+
+
+def prox_newton_solver(
+        X, y, datafit, penalty, w, Xw, max_iter=50, max_epochs=1000, max_backtrack=10,
+        min_pn_cd_epochs=2, max_pn_cd_epochs=10, p0=10, tol=1e-4, verbose=0):
+    r"""Run a prox-Newton solver.
+
+    Parameters
+    ----------
+    X : array, shape (n_samples, n_features)
+        Training data.
+
+    y : array, shape (n_samples,)
+        Target values.
+
+    datafit : instance of Datafit class
+        Datafitting term.
+
+    penalty : instance of Penalty class
+        Penalty used in the model.
+
+    w : array, shape (n_features,)
+        Coefficient vector.
+
+    Xw : array, shape (n_samples,)
+        Model fit.
+
+    max_iter : int, optional
+        Maximum number of iterations (definition of working set and
+        resolution of problem restricted to features in working set).
+
+    max_epochs : int, optional
+        Maximum number of prox-Newton epochs on each subproblem.
+
+    max_backtrack : int, optinal
+        Maximum number of backtracking steps for the line search.
+
+    min_pn_cd_epochs : int, optional
+        Minimum number of iterations used in the prox-Newton coordinate
+        descent.
+
+    max_pn_cd_epochs : int, optional
+        Maximum number of iterations used in the prox-Newton coordinate
+        descent.
+
+    p0 : int, optional
+        First working set size.
+
+    tol : float, optional
+        The tolerance for the optimization.
+
+    verbose : bool or int, optional
+        Amount of verbosity. 0/False is silent.
+
+    Returns
+    -------
+    w : array, shape (n_features,)
+        Coefficient vector.
+
+    obj_out : array, shape (n_iter,)
+        The objective values at every outer iteration.
+
+    stop_crit : float
+        Value of stopping criterion at convergence.
+    """
+    if not isinstance(datafit, (Logistic, Logistic_32)):
+        raise ValueError("Prox-Newton solver only supports Logistic datafits.")
+    n_features = X.shape[1]
+    pen = penalty.is_penalized(n_features)
+    unpen = ~pen
+    n_unpen = unpen.sum()
+    obj_out = []
+    all_feats = np.arange(n_features)
+    stop_crit = np.inf  # initialize for case n_iter=0
+
+    accelerator = AndersonAcceleration(K=5)
```

Comment:
Is there a benefit in using AndersonAccelerator with PN ? In my experience, no

Reply:
Yes, I removed it and it was really slow afterwards (https://github.com/scikit-learn-contrib/skglm/pull/32/commits/18b185fbd6b75194e3da18dbd4657c64b203639f).

Reply:
This is weird. PN iterations break the VAR structure in my experience, see Fig 8 https://www.jmlr.org/papers/volume21/19-587/19-587.pdf 

How does the current solver compare to Blitz and Celer-PN ?